src/MongoDB.EntityFrameworkCore/Serializers/EntitySerializer.cs

```diff
@@ -62,6 +62,21 @@ public bool TryGetMemberSerializationInfo(string memberName, out BsonSerializati
             if (property != null)
             {
                 var elementName = property.GetElementName();
+                if (property.IsPrimaryKey())
+                {
+                    // TODO: should rework this once we are ready for proper composite key implementation
+                    // Current implementation:
+                    // - remap single-field PK to be named "_id" as per MondoDB restriction
+                    // - for composite-PK: we create nested document that include all PK fields
+                    //   and also put the fields into the document itself to let LINQ provider translate expressions
+                    var pk = _entityType.FindPrimaryKey();
+
+                    if (pk.Properties.Count == 1 && pk.Properties[0] == property)
+                    {
+                        elementName = "_id";
+                    }
+                }
+
```

Comment:
The PrimaryKeyConvention I checked in now makes sure that _id is set for single-field PKs.

Reply:
Let's delete this commented-out code before check-in.